magenta/models/music_vae/js/index.ts

```diff
@@ -56,74 +49,73 @@ export class Nade {
 
   sample(encBias: dl.Tensor2D, decBias: dl.Tensor2D) {
     const batchSize = encBias.shape[0];
+    return dl.tidy(()=> {
+      let samples: dl.Tensor1D[] = [];
+      let a = encBias.clone();
+
+      for (let i = 0; i < this.numDims; i++) {
+        let h = dl.sigmoid(a);
+        let encWeights_i = dl.slice2d(
+          this.encWeights, [i, 0], [1, this.numHidden]).as1D();
```

Comment:
just so you know, we have a chain API now which passes "this" as the first argument (of course no need to change this if using the TF way is more your style).

This would look like:

```
let encWeights_i = this.encWeights.slice([i, 0], [1, this.numHidden]).as1D();
```

Reply:
The above `dl.sigmoid(a)` becomes `a.sigmoid()`

Reply:
Done, here and elsewhere.